hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/RouterServerUtil.java

```diff
@@ -101,4 +123,26 @@ public static void logAndThrowRunTimeException(String errMsg, Throwable t)
       throw new RuntimeException(errMsg);
     }
   }
+
+  /**
+   * Throws an RunTimeException due to an error.
+   *
+   * @param errMsgFormat the error message format string.
+   * @param t the throwable raised in the called class.
+   * @param args referenced by the format specifiers in the format string.
+   * @throws RuntimeException on failure
+   */
+  @Public
+  @Unstable
+  public static void logAndThrowRunTimeException(String errMsgFormat, Throwable t, Object... args)
```

Comment:
The order of the parameters is tricky.
I wonder if it would be cleaner to have the Throwable first.

Reply:
I will modify the code.